examples/bits.rs

```diff
@@ -1,98 +1,135 @@
+    // draw
+    let right_edge = BIT_BOX * 4.0;
+    let top_edge = BIT_HEIGHT * (text.len() as f64 / 2.0);
+    // The turtle starts from the top right of the region,
+    turtle.go_to((right_edge, top_edge));
+    // and walks left
+    turtle.set_heading(180.0);
 
-/// An implementation of a vector of bits that can be easily set individually
-/// Note that this is meant to be a teaching implementation and so is not necessarily the most
-/// sophisticated or performant way to do everything. The goal is to make the code understandable
-/// and correct.
-/// See the bit-vec crate and others on crates.io for more advanced implementations.
-#[derive(Debug, Clone, Default)]
-pub struct BitVec {
-    bytes: Vec<u8>,
-}
+    // Now let’s draw some data on the screen, as individual bits
 
-impl fmt::Display for BitVec {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        // EXERCISE: Only print the bytes up to the length of bytes that have been pushed, not
-        // the entire capacity. (Similar to how Vec prints its len, not its capacity)
-        for byte in &self.bytes {
-            write!(f, "{:0width$b}", byte, width = BITS)?;
-        }
-        Ok(())
+    // Rust strings can iterate over the bytes used to represent them.
+    for byte in text.bytes() {
+        // For each byte (`u8`), we use `bitvec` to make a view into its bits.
+        //
+        // The `LittleEndian` means that the view moves from right to left
+        // across the byte, which fits how our turtle moves from right to left
+        // across the screen.
+        let bits = byte.bits::<LittleEndian>();
+        // Then we draw the byte’s bits as a row
+        draw_row(&mut turtle, bits);
+        // And reset the turtle to the right edge.
+        turtle.set_x(right_edge);
     }
-}
 
-impl BitVec {
-    /// Create a new BitVec with the given size in bits
-    /// Note that the actual size of the vector is in bytes so `bits` will be rounded up to the
-    /// nearest multiple of 8 >= bits.
-    pub fn with_capacity(bits: usize) -> Self {
-        if bits > 0 {
-            // Ceiling division of bits / BITS so that we get >= bits capacity
-            // From: https://stackoverflow.com/a/2745086/551904
-            let capacity = 1 + ((bits - 1) / BITS);
-            Self { bytes: vec![0; capacity] }
-        } else {
-            Default::default()
-        }
-    }
+    // `bitvec` can look at more than just `u8`. Let’s try looking at the bits
+    // that represent a number!
 
-    /// Returns the size in bits of the array
-    pub fn len(&self) -> usize {
-        self.len_bytes() * BITS
-    }
+    // First we need to pick a number. Rust provides some interesting numbers
+    // for us.
+    let tau = std::f64::consts::PI * 2.0;
+    // But `bitvec` doesn’t know how to view all numbers. The standard library
+    // provides a function `to_bits(f64) -> u64`, which turns it into a number
+    // `bitvec` does know how to view.
```

Comment:
Maybe a link to how fp numbers are represented so there's some context for understanding the bits?

Reply:
How about [this one](https://en.wikipedia.org/wiki/Double-precision_floating-point_format):

```diff
    //
+     // The `f64` data type has a memory encoding standardized by the IEEE-754
+     // document. You can read more about it here:
+     // https://en.wikipedia.org/wiki/Double-precision_floating-point_format
```

Reply:
Perfect!